cosmos/providers/databricks/plugin.py

```diff
@@ -0,0 +1,279 @@
+    """Constructs a link to monitor a Databricks Job Run."""
+
+    name = "See Databricks Job Run"
+
+    def get_link(
+        self,
+        operator: BaseOperator,
+        dttm=None,
+        *,
+        ti_key: TaskInstanceKey | None = None,
+    ) -> str:
+        dag = get_airflow_app().dag_bag.get_dag(ti_key.dag_id)
+        dag.get_task(ti_key.task_id)
+        # Should we catch the exception here if there is no return value?
+        metadata = XCom.get_one(
+            task_id=ti_key.task_id,
+            dag_id=ti_key.dag_id,
+            run_id=ti_key.run_id,
+            key="return_value",
+        )
+        hook = DatabricksHook(metadata.databricks_conn_id)
+        return f"https://{hook.host}/#job/{metadata.databricks_job_id}/run/{metadata.databricks_run_id}"
+        return f"/foo/bar?dag_id={operator.dag_id}&task_id={operator.task_id}"
+
+
+class DatabricksJobRepairAllFailedLink(BaseOperatorLink):
+    """Constructs a link to send a request to repair all failed databricks tasks."""
+
+    name = "Repair All Failed Tasks"
+
+    def get_link(
+        self,
+        operator,
+        dttm=None,
+        *,
+        ti_key: TaskInstanceKey | None = None,
+    ) -> str:
+        # Should we catch the exception here if there is no return value?
+        metadata = XCom.get_one(
+            task_id=ti_key.task_id,
+            dag_id=ti_key.dag_id,
+            run_id=ti_key.run_id,
+            key="return_value",
+        )
+
+        tasks_str = self.get_tasks_to_run(ti_key, operator)
+        print(f"tasks to rerun {tasks_str}")
+        return (
+            f"/repair_databricks_job?dag_id={ti_key.dag_id}&"
+            f"databricks_conn_id={metadata.databricks_conn_id}&"
+            f"databricks_run_id={metadata.databricks_run_id}&"
+            f"tasks_to_repair={tasks_str}"
+        )
+
+    def get_tasks_to_run(self, ti_key: TaskInstanceKey, operator: BaseOperator) -> str:
+        dag = get_airflow_app().dag_bag.get_dag(ti_key.dag_id)
+        dr = self.get_dagrun(dag, ti_key.run_id)
+        failed_and_skipped_tasks = self._get_failed_and_skipped_tasks(dr)
+        tasks_to_run = {
+            ti: t
+            for ti, t in operator.task_group.children.items()
+            if ti in failed_and_skipped_tasks
+        }
+        tasks_str = ",".join(
+            get_databricks_task_ids(operator.task_group.group_id, tasks_to_run)
+        )
+        return tasks_str
+
+    def _get_failed_and_skipped_tasks(self, dr: DagRun):
+        """
+        Returns a list of task IDs for tasks that have failed or have been skipped in the given DagRun.
+
+        :param dr: The DagRun object for which to retrieve failed and skipped tasks.
+
+        :return: A list of task IDs for tasks that have failed or have been skipped.
+        """
+        return [
+            t.task_id
+            for t in dr.get_task_instances(
+                state=["failed", "skipped", "up_for_retry", "upstream_failed"],
```

Comment:
Do we need `up_for_retry` here? Do we see any possible race conditions here? e.g. A task that was up for retry gets retried by the Airflow scheduler just before a repair is triggered from here.

Reply:
Yes we need those too because we need to clear those tasks for them to actually be retried. If anything we might want to advise people to turn off retries for databricks taskgroup jobs or default the tasks to not retry. @tatiana @kaxil WDYT?

Reply:
Do we really need `up_for_retry` here, @dimberman ?

Reply:
@tatiana yes, because we can't use Airflow's retries so any task that is "up_for_retry" will just continue to fail unless it is added to the repair job.